### PR TITLE
remove event and env from http span

### PR DESCRIPTION
**Pull request by LeonPev**

### CODE REVIEW

File: internal/transform/span.go [hunk 1]
```diff
@@ -108,6 +124,8 @@ func (m *mapper) Transform(invocationStartedTimestamp int64) telemetry.Span {
 
 		if isStartSpan {
 			lumigoSpan.ID = fmt.Sprintf("%s_started", lumigoSpan.ID)
+			deadline, _ := m.ctx.Deadline()
```
Comment (nirLumigo) [inline internal/transform/span.go:125]: What does it mean?

File: internal/transform/span.go [hunk 2]
```diff
@@ -239,8 +238,6 @@ func (m *mapper) getHTTPInfo(attrs map[string]interface{}) *telemetry.SpanHttpIn
 
 	if reqBody, ok := attrs["http.request_body"]; ok {
 		spanHttpInfo.Request.Body = fmt.Sprint(reqBody)
-	} else {
-		m.logger.Error("unable to fetch HTTP request body")
```
Comment (nirLumigo) [inline internal/transform/span.go:243]: Why did you delete it?

File: internal/transform/span_test.go [hunk 3]
```diff
@@ -359,9 +356,11 @@ func TestTransform(t *testing.T) {
 		if lumigoSpan.LambdaType == "http" {
 			lumigoSpan.ID = mockLambdaContext.AwsRequestID
 		}
-		if !reflect.DeepEqual(lumigoSpan, tc.expect) {
-			t.Errorf("%s: %#v != %#v", tc.testname, lumigoSpan, tc.expect)
+		// assert.Equal(t, tc.expect, lumigoSpan)
```
Comment (nirLumigo) [inline internal/transform/span_test.go:394]: If it isn't used then we can delete it

File: internal/transform/span_test.go [hunk 4]
```diff
@@ -359,9 +356,11 @@ func TestTransform(t *testing.T) {
...
+		if diff := cmp.Diff(tc.expect, lumigoSpan); diff != "" {
+			t.Errorf("%s mismatch (-want +got):\n%s", tc.testname, diff)
```
Comment (nirLumigo) [inline internal/transform/span_test.go:396]: What's the difference between the previous section?

File: wrapper_test.go [hunk 5]
```diff
@@ -203,30 +203,30 @@ func (w *wrapperTestSuite) TestLambdaHandlerE2ELocal() {
...
+		// 	},
+		// },
+		// {
+		// 	name:     "input: struct event, return error",
+		// 	input:    9090,
+		// 	expected: expected{"", errors.New("failed error")},
+		// 	handler: func(event int) (*struct{ Port int }, error) {
+		// 		return nil, errors.New("failed error")
+		// 	},
+		// },
```
Comment (nirLumigo) [inline wrapper_test.go]: Why did you delete all of these tests?

File: wrapper_test.go [hunk 6]
```diff
@@ -390,5 +390,4 @@ func (w *wrapperTestSuite) TestCheckFailWriteSpanHandler_FailLoadConfig() {
 	assert.Equal(w.T(), 1, len(dirEntries))
 
 	assert.Equal(w.T(), "balagan_stop", dirEntries[0].Name())
-
 }
```
Comment (nirLumigo) [inline wrapper_test.go:393]: I think we also need a test that verifies that function-span doesn't have the httpInfo.

File: internal/transform/span.go (hunk 1 above)
Comment (LeonPev) [reply internal/transform/span.go:125]: this is used to calculate the end value of the start span

File: internal/transform/span.go (hunk 2 above)
Comment (LeonPev) [reply internal/transform/span.go:243]: because not all http traffic has body like GET

File: internal/transform/span_test.go (hunk 4 above)
Comment (LeonPev) [reply internal/transform/span_test.go:396]: this is more descriptive and works as expected (brought to the world by google)

File: wrapper_test.go (hunk 6 above)
Comment (LeonPev) [reply wrapper_test.go:393]: will do

File: go.mod [hunk 7]
```diff
@@ -5,13 +5,13 @@ go 1.17
 require (
 	github.com/aws/aws-lambda-go v1.27.0
 	github.com/aws/aws-sdk-go-v2 v1.16.1
+	github.com/google/go-cmp v0.5.7
```
Comment (nirLumigo) [inline go.mod:8]: Can it be a dev dependency?

File: wrapper_test.go (hunk 6 above)
Comment (LeonPev) [reply wrapper_test.go:393]: this happens here: internal/transform/span_test.go in TestTransform

File: go.mod (hunk 7 above)
Comment (LeonPev) [reply go.mod:8]: this is handled by go build
"When compiling packages, build ignores files that end in '_test.go'."
https://pkg.go.dev/cmd/go#hdr-The_main_module_and_the_build_list

File: internal/transform/span.go [hunk 8]
```diff
@@ -85,14 +85,32 @@ func (m *mapper) Transform(invocationStartedTimestamp int64) telemetry.Span {
...
+		if isWarmStart == "" && !isProvisionConcurrencyInitialization() {
+			lumigoSpan.LambdaReadiness = "cold"
+		} else {
+			lumigoSpan.LambdaReadiness = "warm"
+		}
 	}
 	lumigoSpan.LambdaType = lambdaType
-
+	if isStartSpan {
+		lumigoSpan.LambdaEnvVars = m.getEnvVars()
```
Comment (nirLumigo) [inline internal/transform/span.go:112]: We send the env vars on both start-span and end-span, all the data from the start-span should also be on the end-span.
So maybe it should be `if isFunctionSpan`

**Review by nirLumigo — CHANGES_REQUESTED** [review]
